### PR TITLE
📝 Update API docs to include 400, 404 responses

### PR DESCRIPTION
**Pull request by znatty22**
Addresses #192

### CODE REVIEW

File: dataservice/utils.py [hunk 1]
```diff
@@ -13,3 +14,14 @@ def iterate_pairwise(iterable):
     a, b = tee(iterable)
     next(b, None)
     return zip(a, b)
+
+
+def to_snake_case(camel_case_str):
```
Comment (dankolbman) [inline dataservice/utils.py:19]: Our resources are hyphenated, not underscored. eg `/genomic-files`, `/family-relationships`

Comment (znatty22) [reply dataservice/utils.py:19]: Yea our endpoints are hyphenated but the error messages contain our entity names which are snake cased. Example: 
```json
{
    "_status": {
        "code": 400,
        "message": "could not create family_relationship: {'participant_id': ['Missing data for required field.'], 'participant_to_relative_relation': ['Missing data for required field.'], 'relative_id': ['Missing data for required field.']}"
    }
}
```

Comment (dankolbman) [reply dataservice/utils.py:19]: Right, but the snake case is still being used for the resource names in the pagination

File: dataservice/api/common/schemas.py [hunk 2]
```diff
@@ -97,12 +112,13 @@ class RespSchema(Schema):
...
 
-def paginated_generator(schema):
+def paginated_generator(resource_name, schema):
     class PaginatedSchema(Schema):
         _status = fields.Dict(example={'message': 'success', 'code': 200})
-        _links = fields.Dict(example={'next': '/participants?after=1519402953',
-                                      'self': '/participants?after=1519402952'
-                                      })
+        _links = fields.Dict(
+            example={'next': '/{}?after=1519402953'.format(resource_name),
```
Comment (dankolbman) [inline dataservice/api/common/schemas.py]: might want to do a `resource_name.replace('_', '-')` here

Comment (znatty22) [reply dataservice/api/common/schemas.py]: Oh i see, 👍

File: dataservice/api/common/schemas.py [hunk 3]
```diff
@@ -81,14 +81,29 @@ class ErrorSchema(Schema):
...
-        fields = ('description', 'code')
+        fields = ("description", "code")
 
     @post_dump(pass_many=False)
     def wrap_envelope(self, data):
         return {'_status': {'message': data['description'],
                             'code': data['code']}}
 
 
+def error_response_generator(resource_name, status_code):
```
Comment (dankolbman) [inline dataservice/api/common/schemas.py:119]: I'm not sure we should regenerate an identical schema for every error type. It seems to violate the idea of polymorphism, which swagger supports.

File: dataservice/api/common/views.py [hunk 4]
```diff
@@ -38,12 +41,26 @@ def register_spec(cls, spec):
...
             for name, schema in c.schemas.items():
+                # Entity response schemas
                 spec.definition(name, schema=schema)
                 ResponseSchema = response_generator(schema)
                 spec.definition(name + 'Response', schema=ResponseSchema)
-                PaginatedSchema = paginated_generator(schema)
-                spec.definition(name + 'Paginated', schema=PaginatedSchema)
+
+                # Pagination schemas
+                if c.__name__.endswith('ListAPI'):
```
Comment (dankolbman) [inline dataservice/api/common/views.py]: Maybe better to use the url rule?

Comment (znatty22) [reply dataservice/api/common/views.py]: Wasn't sure if I should do that or use the ListAPI postfix bc the url rules for paginated endpoints might not always be /\<entity\>. For example we might have /genomic-files?sequence_center=<kf_id> soon. Figured looking for the 'ListAPI' postfix might be more explicit and a good assumption for paginated endpoints?

Comment (dankolbman) [reply dataservice/api/common/views.py]: I don't think the query params show up in url rules?

**Review by dankolbman — APPROVED** [review]

**Review by parimalak — APPROVED** [review]
👍

**Pull request merged**